src/rules/maxLineLengthRule.ts

```diff
@@ -44,20 +69,52 @@ export class Rule extends Lint.Rules.AbstractRule {
         return `Exceeds maximum line length of ${lineLimit}`;
     }
 
-    public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
-    }
-
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments));
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (
+            line.contentLength > ctx.options.limit &&
+            !(
+                ctx.options.ignoreUrls &&
+                hasUrl(line, ctx.sourceFile) &&
+                lineLengthMinusUrl(line, ctx.sourceFile) < ctx.options.limit
+            )
+        ) {
+            ctx.addFailureAt(
+                line.pos,
+                line.contentLength,
+                Rule.FAILURE_STRING_FACTORY(ctx.options.limit),
+            );
         }
     }
 }
+
+function hasUrl(line: LineRange, sourceFile: ts.SourceFile): boolean {
+    return urlRegex().test(sourceFile
+                            .text.substr(line.pos, line.contentLength));
+}
+
+function isOptions(arg: number | Options): arg is Options {
+    return (arg as Options).limit !== undefined;
+}
+
+function lineLengthMinusUrl(line: LineRange, sourceFile: ts.SourceFile): number {
+    return sourceFile
+            .text
+            .substr(line.pos, line.contentLength)
+            .replace(urlRegex(), "")
+            .length;
+}
+
+function parseOptions(args: number[] | Options[]): Options {
+    return isOptions(args[0])
+        ? args[0] as Options
```

Comment:
you should apply default values here as well, because the user can also provide only `Partial<Options>`